activerecord/lib/active_record/database_configurations.rb

```diff
@@ -93,7 +91,7 @@ def empty?
     private
       def env_with_configs(env = nil)
         if env
-          configurations.select { |db_config| db_config.env_name == env }
+          configurations.select { |db_config| db_config.env_name == env.to_s }
```

Comment:
Can you undo the changes that allow non-strings? I don't want the configs_for api to accept symbols. Maybe I'll change my mind later on but this API is a WIP so I'd rather just make it relatively strict for now.